pkg/pod/admission/limit_request.go

```diff
@@ -0,0 +1,98 @@
+package admission
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/ghodss/yaml"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+func init() {
+	admission.RegisterPlugin("PodLimitRequest", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newPodLimitRequest(config)
+	})
+}
+
+type limitRequestConfig struct {
+	Enabled                   bool    `json:"enabled"`
+	LimitCpuToMemoryRatio     float64 `json:"limitCpuToMemoryRatio"`
+	CpuRequestToLimitRatio    float64 `json:"cpuRequestToLimitRatio"`
+	MemoryRequestToLimitRatio float64 `json:"memoryRequestToLimitRatio"`
+}
+type podLimitRequest struct {
+	*admission.Handler
+	limitRequestConfig
+}
+
+// newPodLimitRequest returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newPodLimitRequest(config io.Reader) (admission.Interface, error) {
+	parsed, err := ReadConfig(config)
+	return &podLimitRequest{
+		Handler:            admission.NewHandler(admission.Create, admission.Update),
+		limitRequestConfig: parsed,
+	}, err
+}
+
+func ReadConfig(config io.Reader) (limitRequestConfig, error) {
+	target := limitRequestConfig{}
+	buffer := new(bytes.Buffer)
+	buffer.ReadFrom(config)
+	err := yaml.Unmarshal(buffer.Bytes(), &target)
+	if err == nil && target.Enabled {
+		if target.LimitCpuToMemoryRatio == 0.0 && target.CpuRequestToLimitRatio == 0.0 && target.MemoryRequestToLimitRatio == 0.0 {
+			return target, fmt.Errorf("PodLimitRequest plugin enabled but no ratios specified")
+		}
+		if target.LimitCpuToMemoryRatio < 0.0 {
+			return target, fmt.Errorf("LimitCpuToMemoryRatio must be positive")
+		}
+		if target.CpuRequestToLimitRatio < 0.0 || target.CpuRequestToLimitRatio > 1.0 {
+			return target, fmt.Errorf("CpuRequestToLimitRatio must be between 0.0 and 1.0")
+		}
+		if target.MemoryRequestToLimitRatio < 0.0 || target.MemoryRequestToLimitRatio > 1.0 {
+			return target, fmt.Errorf("MemoryRequestToLimitRatio must be between 0.0 and 1.0")
+		}
+	}
+	return target, err
+}
+
+func (a *podLimitRequest) Admit(attr admission.Attributes) error {
+	if !a.Enabled {
+		return nil
+	}
+	pod, ok := attr.GetObject().(*kapi.Pod)
```

Comment:
subresource API calls come through admission as well. in some cases (like posting to `pods/foo/status`, the API object send to the endpoint is actually a `*api.Pod` (so the type check above would succeed).

I'm pretty sure this admission controller should only be paying attention to creates on pods (not to updates, and not to subresources). You can check that by returning early if `len(attr.GetSubresource()) > 0`, among other ways